### PR TITLE
Allow adding storage locations and setup permissions 

### PR DESCRIPTION
**Pull request by curtishall**

### CODE REVIEW

File: www/ajax/storagecheck.php [hunk 1]
```diff
@@ -21,47 +20,31 @@ public function postData()
...
+        return array('F', $output);
+    } elseif (strpos($output, 'Changing permissions') !== false || strpos($output, 'Modifying permissions and ownership') !== false) {
+        return array('OK', 'Permissions and ownership changed for ' . $path);
+    } elseif (strpos($output, 'already has the correct permissions and ownership') !== false) {
+        return array('OK', '');
+    } elseif (strpos($output, 'does not exist. Creating it now') !== false) {
+        // Handle the case where the directory is being created
+        return array('OK', 'Directory created and permissions set for ' . $path);
+    } else {
+        return array('F', 'Unexpected output from shell script: ' . $output);
```
Comment (andrey-utkin) [inline www/ajax/storagecheck.php:45]: As I start typing the storage path to add, I keep getting red popups "Unexpected output from shell script: " (with nothing after). This seems to happen even for complete, existing, accessible paths.

Saving additional location fails with the same popups. No reason given. Nothing in the logs (neither /var/log/bluecherry.log, nor /var/log/nginx/bluecherry-error.log.

**Review by andrey-utkin — CHANGES_REQUESTED** [review]
Could you please clarify the intended effect of the changes felt by the user?

I tested this in runtime and the UI fails to add a location. I am yet to compare this with a build off master branch.

**Comment by curtishall** [comment]
@andrey-utkin Code works on https://192.168.86.66:7001/storage (focal, latest from this branch)

Can you test there?  Perhaps we have a permission issue with the script?

File: www/ajax/storagecheck.php (hunk 1 above)
Comment (curtishall) [reply www/ajax/storagecheck.php:45]: this file was missing from install
/usr/share/bluecherry/scripts/check_dir_permission.sh

**Review by andrey-utkin — APPROVED** [review]
Looks good to me

**Pull request merged**